rpc/mocks_test.go

```diff
@@ -0,0 +1,208 @@
+package rpc
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/athanorlabs/atomic-swap/common"
+	"github.com/athanorlabs/atomic-swap/common/types"
+	mcrypto "github.com/athanorlabs/atomic-swap/crypto/monero"
+	"github.com/athanorlabs/atomic-swap/net"
+	"github.com/athanorlabs/atomic-swap/net/message"
+	"github.com/athanorlabs/atomic-swap/protocol/swap"
+	"github.com/athanorlabs/atomic-swap/protocol/txsender"
+)
+
+//
+// This file only contains mock definitions used by other test files
+//
```

Comment:
These mocks were previously located in `ws_test.go`.  I didn't realize that they were also being shared with `net_test.go`, so I thought pulling them out into a separate file might make make it clear that they are shared. I put `panic`s in most of them, to make it more obvious to the reader which functions are actually getting invoked.  If we decide to instrument the mocks with testify's mocking at a later time (https://github.com/stretchr/testify#mock-package), we wouldn't need to instrument all functions.

Reply:
it's a lot more complex to update the tests to not use mocking, as we basically need to start up everything in the daemon. it might be nice to add them as integration tests, instead of unit tests.